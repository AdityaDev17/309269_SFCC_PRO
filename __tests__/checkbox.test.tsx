import Checkbox from "../src/components/atomic/Checkbox/Checkbox";
import { fireEvent, render, screen } from "@testing-library/react";

describe('Checkbox Component', () => {
    test("renders without crashing", () => {
        render(<Checkbox data-testid="checkbox"/>);
        const checkbox = screen.getByTestId("checkbox");
        expect(checkbox).toBeInTheDocument();
    });

    test("toggle check state when clicked", () => {
        render(<Checkbox data-testid="checkbox"/>);
        const checkbox = screen.getByTestId("checkbox");
        expect(checkbox).not.toBeChecked();
        fireEvent.click(checkbox);
        expect(checkbox).toBeChecked();
    });
    
    test("matches snapshot", () => {
        const {container} = render(<Checkbox/>)
        expect(container).toMatchSnapshot();
    })
})