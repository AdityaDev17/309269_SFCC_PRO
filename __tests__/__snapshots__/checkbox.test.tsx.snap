// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component matches snapshot 1`] = `
<div>
  <button
    aria-checked="false"
    class="Checkbox"
    data-state="unchecked"
    role="checkbox"
    type="button"
    value="on"
  />
</div>
`;
